### PR TITLE
Refactor integration tests using ShapleQTestContext structure 

### LINKED ISSUE #150: Refactor integration tests
### Issues
* current integration tests are too messy to understand the purpose of a test

### Todo
* Function bodies for integration test should be more simplified to clarify the purpose of the test
* Define custom test-context which handles life cycles for the broker, zk and clients
* Encapsulate goroutines for subscribing and publishing functionalities to simplify the body of test case

### PR DESCRIPTION
**Pull request by elon0823**
close #150 

### Test dependency
This PR is separated from #148 to make that PR smaller, which means some tests are bound by the implementations from that PR. I had performed all tests on the #148's branch and checked all test is passed. I suggest the reviewers to focus on the structure of test-context(`integration_test.go`) and much much clearer test body(`stream_test.go`, `rpc_test.go`) itself.

### CODE REVIEW

**Comment by elon0823** [comment]
Functional tests(`TestMultiFragmentsTotalConsume`, `TestMultiFragmentsOptionalConsume`) for #148 is added to this PR.

File: integration_test/stream_test.go [hunk 1]
```diff
@@ -285,337 +79,210 @@ func TestPubSub(t *testing.T) {
...
-
-					if len(subscribedRecords) == len(totalPublishedRecords) {
-						fmt.Printf("done %s\n", name)
+		}).
+		asyncPublish(expectedRecords)
+
+	// setup consumer for each fragment
+	for fragmentId, offset := range testParams.fragmentOffsets {
+		wg.Add(1)
+		func(fid uint32, startOffset uint64) {
```
Comment (1dennispark) [inline integration_test/stream_test.go:252]: Why use lambda in here? May I have leaked go knowledge?

**Review by 1dennispark — COMMENTED** [review]
Good tests. The intent of testing case is good.

How about adding test framework of it?

File: integration_test/stream_test.go (hunk 1 above)
Comment (elon0823) [reply integration_test/stream_test.go:252]: to capture the values inside

Comment (elon0823) [reply integration_test/stream_test.go:252]: It can just replaced to new variable assignment.. I'll fix it

**Review by 1dennispark — APPROVED** [review]

**Pull request merged**